### PR TITLE
Node: Remove Neon support

### PR DESCRIPTION
**Pull request by bruce-riley**
This PR removes guardian support for Neon, which was only ever deployed to testnet.

This PR also removes an obsolete readiness unit test.

### CODE REVIEW

**Review by evan-gray — APPROVED** [review]
On the fence about removing the chain id, but agree it wasn't used in mainnet.

File: node/pkg/proto/publicrpc/v1/publicrpc.pb.go [hunk 1]
```diff
@@ -25,24 +25,24 @@ const (
...
+	ChainID_CHAIN_ID_ALGORAND    ChainID = 8
+	ChainID_CHAIN_ID_AURORA      ChainID = 9
+	ChainID_CHAIN_ID_FANTOM      ChainID = 10
+	ChainID_CHAIN_ID_KARURA      ChainID = 11
+	ChainID_CHAIN_ID_ACALA       ChainID = 12
+	ChainID_CHAIN_ID_KLAYTN      ChainID = 13
+	ChainID_CHAIN_ID_CELO        ChainID = 14
+	ChainID_CHAIN_ID_NEAR        ChainID = 15
+	ChainID_CHAIN_ID_MOONBEAM    ChainID = 16
+	// OBSOLETE: CHAIN_ID_NEON = 17;
```
Comment (SEJeff) [inline node/pkg/proto/publicrpc/v1/publicrpc.pb.go:45]: I was thinking when I first pulled this PR up to check to see if you did this. Nice.

I'd be half tempted to make a unit test to ensure that ChainID 17 is never used. Maybe use the machinery that converts the chainId to a string and if it returns anything then fail the test or something.

File: sdk/vaa/structs.go [hunk 2]
```diff
@@ -427,8 +422,6 @@ const (
 	ChainIDNear ChainID = 15
 	// ChainIDMoonbeam is the ChainID of Moonbeam
 	ChainIDMoonbeam ChainID = 16
-	// ChainIDNeon is the ChainID of Neon
-	ChainIDNeon ChainID = 17
```
Comment (SEJeff) [inline sdk/vaa/structs.go:431]: You might add a comment here as well for the obsolete chain id 17.

**Review by SEJeff — APPROVED** [review]

File: sdk/vaa/structs.go (hunk 2 above)
Comment (bruce-riley) [reply sdk/vaa/structs.go:431]: Added a comment saying it's obsolete.

**Review by SEJeff — APPROVED** [review]

**Review by panoel — APPROVED** [review]

**Review by evan-gray — APPROVED** [review]

**Pull request merged**